libs/cdk/render-strategies/src/lib/rx-schedule-task.ts

```diff
@@ -0,0 +1,47 @@
+import { NgZone } from '@angular/core';
+import {
+  cancelCallback,
+  scheduleCallback,
+  PriorityLevel,
+} from '@rx-angular/cdk/internals/scheduler';
+import { RxConcurrentStrategyNames } from './model';
+
+type StrategiesPriorityRecord = Record<
+  RxConcurrentStrategyNames,
+  PriorityLevel
+>;
+
+const strategiesPrio: StrategiesPriorityRecord = {
+  immediate: PriorityLevel.ImmediatePriority,
+  userBlocking: PriorityLevel.UserBlockingPriority,
+  normal: PriorityLevel.NormalPriority,
+  low: PriorityLevel.LowPriority,
+  idle: PriorityLevel.IdlePriority,
+};
+
+/**
+ * @description
+ * This function is used to schedule a task with a certain priority.
+ * It is useful for tasks that can be done asynchronously.
+ *
+ * ```ts
+ * const task = rxScheduleTask(() => localStorage.setItem(state, JSON.stringify(state));
+ * ```
+ */
+export const rxScheduleTask = (
+  work: (...args: any[]) => void,
+  strategy: keyof StrategiesPriorityRecord = 'normal',
```

Comment:
How would a user use the function with a custom strategy? the typing wouldn't allow it currently. Or am I missing something?

Reply:
I guess we should not use custom strategy here, because only thing we need from strategy is priority. And scheduler not working with custom priority levels.

That's why we need this, to provide users familiar API instead of `PriorityLevel` enum.

```
const strategiesPrio: StrategiesPriorityRecord = {
  immediate: PriorityLevel.ImmediatePriority,
  userBlocking: PriorityLevel.UserBlockingPriority,
  normal: PriorityLevel.NormalPriority,
  low: PriorityLevel.LowPriority,
  idle: PriorityLevel.IdlePriority,
};
```